src/network/socket.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "network/ip_address.h"
+
+namespace lightstep {
+/**
+ * Wraps a system socket structure.
+ */
+class Socket {
+ public:
+  Socket();
+
+  Socket(int family, int type);
+
+  explicit Socket(int file_descriptor) noexcept;
+
+  Socket(const Socket&) = delete;
+
+  Socket(Socket&& other) noexcept;
+
+  ~Socket() noexcept;
+
+  Socket& operator=(Socket&& other) noexcept = delete;
+
+  Socket& operator=(const Socket&) = delete;
+
+  /**
+   * @return the file descriptor for this socket.
+   */
+  int file_descriptor() const noexcept { return file_descriptor_; }
+
+  /**
+   * Makes the socket non-blocking.
+   */
+  void SetNonblocking();
+
+  /**
+   * makes the socket blocking.
+   */
+  void SetBlocking();
```

Comment:
(When would we use blocking DNS resolution?)

Reply:
Removed.